### PR TITLE
Kbc 1937 use lib for retry

### PR DESCRIPTION
**Pull request by fejjo**
Jira: KBC-1937

Before asking for review make sure that:

- [x] New client method(s) has tests
- [x] Apiary file is updated
- [x] You declared if there is a BC break or not (will affect next release of a client)

---

### CODE REVIEW

**Comment by fejjo** [comment]
Este to tagnem, ale na review to imho moze ist

**Comment by fejjo** [comment]
@martinjunger pridal som este na review teba keby mas nahodou cas jirka myslim musel tedaz ist prec :)

File: tests/StorageApiTestCase.php [hunk 1]
```diff
@@ -827,17 +833,18 @@ protected function listEvents(Client $client, $eventName, $expectedObjectId = nu
...
-                $retries--;
-                usleep(250 * 1000);
-            } else {
-                break;
-            }
-        }
-        return $events;
+
+            $this->assertNotEmpty($events, 'There were no events');
+            $this->assertEquals($eventName, $events[0]['event'], sprintf('Event does not match'));
```
Comment (jirkasemmler) [inline tests/StorageApiTestCase.php:844]:
```suggestion
            $this->assertEquals($eventName, $events[0]['event'], 'Event does not match');
```

File: tests/StorageApiTestCase.php [hunk 2]
```diff
@@ -9,11 +9,17 @@
 
 namespace Keboola\Test;
 
+use ErrorException;
```
Comment (jirkasemmler) [inline tests/StorageApiTestCase.php:12]: ErrorException, Keboola\StorageApi\Exception a Retry\Policy\CallableRetryPolicy jsou nepouzite

File: tests/StorageApiTestCase.php (hunk 1 above)
Comment (jirkasemmler) [reply tests/StorageApiTestCase.php:844]: nebo nevim proc by tam ten sprintf mel byt

**Review by jirkasemmler — CHANGES_REQUESTED** [review]
malickosti, ta trida ComponentEvetsTest mi prochazi, takze cajk :)

File: tests/StorageApiTestCase.php (hunk 2 above)
Comment (fejjo) [reply tests/StorageApiTestCase.php]: Supr dik to som si nevsimol ze som tam nechal

**Review by jirkasemmler — APPROVED** [review]

File: tests/StorageApiTestCase.php (hunk 1 above)
Comment (tomasfejfar) [reply tests/StorageApiTestCase.php]: Měl tam být aby tam mohlo být `no event matches 'storage.configurationCreated'` 😝

Comment (fejjo) [reply tests/StorageApiTestCase.php]: Jo jasne dik :)

**Review by jirkasemmler — APPROVED** [review]
cool, double palecek

**Comment by fejjo** [comment]
rebase

**Pull request merged**